packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -568,26 +577,39 @@ String? _buildWhereQuery({
   Expression? manyRelationWhereAddition,
   _SubQueries? subQueries,
   _ListQueryAdditions? listQueryAdditions,
+  bool wrapWhereInNot = false,
 }) {
-  List<String> whereQuery = [];
+  var whereQuery = '';
 
   if (where != null) {
-    whereQuery.add(_resolveWhereQuery(where: where, subQueries: subQueries));
+    whereQuery += _resolveWhereQuery(where: where, subQueries: subQueries);
+
+    if (wrapWhereInNot) {
+      whereQuery = 'NOT $whereQuery';
+    }
   }
 
   if (listQueryAdditions != null) {
-    whereQuery.add('${listQueryAdditions.whereAddition}');
+    if (whereQuery.isNotEmpty) {
+      whereQuery += ' AND ';
+    }
+
+    whereQuery += '${listQueryAdditions.whereAddition}';
   }
 
   if (manyRelationWhereAddition != null) {
-    whereQuery.add('$manyRelationWhereAddition');
-  }
+    if (whereQuery.isNotEmpty) {
+      if (manyRelationWhereAddition is EveryExpression) {
+        whereQuery += ' OR ';
+      } else {
+        whereQuery += ' AND ';
+      }
+    }
 
-  if (whereQuery.isEmpty) {
-    return null;
+    whereQuery += '$manyRelationWhereAddition';
   }
 
-  return whereQuery.join(' AND ');
+  return whereQuery.isEmpty ? null : whereQuery;
```

Comment:
Suggestion: Make the whereQuery null by default instead, this makes it more obvious that it can be "empty" we can compile time checks and we can simply return it without doing this check here.

Reply:
Couldn't make it null but worked on reducing the footprint of some of the if statements.